buildci.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+function prevent_timeout() {
+    local i=0
+    while [ -e /proc/$1 ]; do
+        # print zero width char every 3 minutes while building
+        if [ "$i" -eq "180" ]; then printf %b '\u200b'; i=0; else i=$((i+1)); fi
+        sleep 1
+    done
+}
+
+function print_reactor_summary() {
+    sed -ne '/\[INFO\] Reactor Summary:/,$ p' "$1" | sed 's/\[INFO\] //'
+}
+
+function mvnp() {
+    set -o pipefail # exit build with error when pipes fail
+    local command=(mvn $@)
+    exec "${command[@]}" 2>&1 | # execute, redirect stderr to stdout
+	stdbuf -o0 grep -vE "Download(ed|ing) from [a-z.]+: https:" | # filter out downloads
+        tee .build.log | # write output to log
+        stdbuf -oL grep -E '^\[INFO\] Building .+ \[.+\]$' | # filter progress
+        stdbuf -o0 sed -uE 's/^\[INFO\] Building (.*[^ ])[ ]+\[([0-9]+\/[0-9]+)\]$/\2| \1/' | # prefix project name with progress
+        stdbuf -o0 sed -e :a -e 's/^.\{1,6\}|/ &/;ta' & # right align progress with padding
+    local pid=$!
+    prevent_timeout $pid &
+    wait $pid
+}
+
+# Determine if this is a new addon -> Perform tests + integration tests and all SAT checks with increased warning level
+CHANGED_DIR=`git diff --diff-filter=A --dirstat=files,0 master...HEAD bundles/ | sed 's/^[ 0-9.]\+% bundles\///g' | grep -o -P "^([^/]*)" | uniq`
```

Comment:
Should we also have a special build for people changing only files in a single module, now it only focuses on either new contributions or it does the complete build.